src-tauri/src/main.rs

```diff
@@ -17,13 +22,14 @@ fn main() {
 /// 
 /// list all directories in the base directory that have a `README.md` file inside
 #[tauri::command]
-async fn list() -> Vec<String> {
-  let base_path = Path::new(home_dir().unwrap().as_path()).join(".notes");
+fn list(config: tauri::State<'_, Config>) -> Vec<String> {
+  let base_path = config.get_base_path().join(".notes");
   match get_note_names(base_path) {
     Ok(note_names) => return note_names,
     Err(e) => error_handling(e.to_string())
   }
-  return Vec::<String>::new();
+
+  Vec::<String>::new()
```

Comment:
@fersaru: I refactored this, since rust encourages to emit `return` when it's not needed. Maybe we should add static analysis using `clippy` in some PR.

Reply:
I added a separate PR for clippy.